spring-xd-dirt/src/main/java/org/springframework/xd/dirt/plugins/spark/streaming/SparkStreamingMessageConverterSupport.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.xd.dirt.plugins.spark.streaming;
+
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+import org.springframework.integration.channel.AbstractMessageChannel;
+import org.springframework.messaging.converter.CompositeMessageConverter;
+import org.springframework.util.MimeType;
+import org.springframework.xd.dirt.integration.bus.converter.ConversionException;
+import org.springframework.xd.dirt.integration.bus.converter.MessageConverterUtils;
+import org.springframework.xd.dirt.plugins.ModuleConfigurationException;
+import org.springframework.xd.dirt.plugins.stream.ModuleTypeConversionPlugin;
+import org.springframework.xd.dirt.util.ConfigLocations;
+
+/**
+ * Utility class to support message conversion in spark streaming module.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+public class SparkStreamingMessageConverterSupport {
+
+
+	/**
+	 * Configure the data type and message converters to be used by the channel on the payload.
+	 *
+	 * @param mimeType the MIME type configured
+	 * @param channel the message channel to send the converted messages
+	 */
+	protected static void configureMessageConverter(MimeType mimeType, AbstractMessageChannel channel) {
```

Comment:
See my comment here: https://github.com/spring-projects/spring-xd/pull/1438/files#r24163988

I do agree we should try to do this in a more Springy way, but the key challenge is that the ApplicationContext for the MessageConverters must be created on the Spark cluster process. Perhaps we can refactor the plugin in such a way that it can be applied to the module there without having to poke these arbitrary holes in it.